content/vault/v1.21.x/content/docs/deploy/run-as-zcx-cluster.mdx

```diff
@@ -0,0 +1,317 @@
+---
+layout: docs
+page_title: Run Vault on IBM z/OS Container Extensions
+description: >-
+  Step-by-step guide to deploying a secure 3-node Vault cluster inside IBM zCX using HAProxy and TLS.
+---
+
+# Run Vault on IBM z/OS Container Extensions (zCX)
+
+This guide walks through deploying a fully secured 3-node HashiCorp Vault Enterprise cluster
+on IBM z/OS Container Extensions (zCX). The setup uses three independent zCX instances, 
+each running Vault with its own unique IP address. A Layer-4 HAProxy load balancer sits 
+in front to distribute traffic, and end-to-end TLS encryption is applied across all Vault nodes 
+and the load balancer to ensure secure communication throughout the cluster.
+
+![Vault zCX cluster deployment](/img/run-as-zcx-cluster.png)
+
+## Step 1: Container image deployment
+
+1. Pull the official Vault Enterprise container image on all three zCX nodes.
+
+  ```shell-session
+  $ docker pull hashicorp/vault-enterprise
+  ```
+
+1. Verify image integrity.
+
+  ```shell-session
+  $ docker images | grep vault-enterprise
+  ```
+
+## Step 2: Persistent volume creation
+
+Create Docker volumes for configuration and data persistence on each node.
+
+1. Create volume for Vault data storage (Raft backend, audit logs).
+
+  ```shell-session
+  $ docker volume create vault-data
+  ```
+
+1. Create volume for Vault configuration files.
+
+  ```shell-session
+  $ docker volume create vault-config
+  ```
+
+1. Verify volume creation.
+
+  ```shell-session
+  $ docker volume ls | grep vault
+  ```
+
+Volume mount points:
+
+- **`vault-data`**: `/vault/data` (stores Raft snapshots, encrypted storage backend)
+- **`vault-config`**: `/vault/config.d` (HCL configuration files)
+
+## Step 4: Vault configuration file setup
+
+Create Vault configuration on each node with node-specific parameters.
+
+1. Access config volume.
+
+  ```shell-session
+  $ docker run --rm -it -v vault-config:/vault alpine sh
+  ```
```

Comment:
```suggestion
```
Since they need to create multiple files and add the certs as well, we typically break the activities into two distinct steps: creating the files and "installing" the files.